mitxgraders/formulagrader.py

```diff
@@ -107,6 +117,64 @@ def schema_config(self):
             Required('failable_evals', default=0): NonNegative(int)
         })
 
+    Utils = namedtuple('Utils', ['tolerance', 'within_tolerance'])
+
+    def get_comparer_utils(self):
+        """Get the utils for comparer function."""
+        def _within_tolerance(x, y):
+            return within_tolerance(x, y, self.config['tolerance'])
+        return self.Utils(tolerance=self.config['tolerance'],
+                          within_tolerance=_within_tolerance)
+
+    @staticmethod
+    def default_comparer(comparer_params, student_input, utils):
+        """
+        Default comparer function.
+
+        Assumes comparer_params is just the expected answer wrapped in a list.
+        """
+        return all([
+            utils.within_tolerance(comparer_param, student_input)
+            for comparer_param in comparer_params
+            ])
+
+    schema_expect = Schema({
+        Required('comparer_params'): [str],
+        # Functions seem not to be usable as default values, so the default comparer is added later.
+        # https://github.com/alecthomas/voluptuous/issues/340
+        Required('comparer'): is_callable_with_args(3)
+    })
+
+    @classmethod
+    def validate_expect(cls, expect):
+        """
+        Validate the answers's expect key.
+
+        >>> result = FormulaGrader.validate_expect('mc^2')
+        >>> expected = {
+        ... 'comparer_params': ['mc^2'],
+        ... 'comparer': FormulaGrader.default_comparer
+        ... }
+        >>> result == expected
+        True
+        """
+        if isinstance(expect, str):
+            return cls.schema_expect({
+                'comparer': cls.default_comparer,
+                'comparer_params': [expect]
+                })
+
+        try:
+            return cls.schema_expect(expect)
+        except Invalid:
+            # Only raise the detailed error message if author is trying to use comparer.
+            if isinstance(expect, dict) and 'comparer' in expect:
+                raise
+            # Otherwise, be generic.
+            else:
+                raise Invalid("Something's wrong with grader's 'answers' configuration key."
```

Comment:
Needs a space after the final period, else the lines are concatenated without space.